### DISCUSSION TITLE
How to calculate the column width and row height of a BIRT table, how to calculate the row height exported to xlsx, and how to calculate the row height and column width ratio of Excel and BIRT

### DISCUSSION THREAD

**Discussion by 858882847** [eclipse-birt/birt#1248] [Q&A]
How to calculate column width ratio columns and row height ratio columns in Excel and BIRT

**Comment by hvbtup** [comment]
If you want to set the widths/height *dynamically based on your data*, you are on your own.
You can use your own best-effort algorithm and specify the widths/heights  in the table/grid item's `onPrepare` event like this:

    // Assuming colWidths is an array of the calculated column widths.
    for (var i = 0; i < colWidths.length; ++i) {
        if (colWidths[i] !== null) {
            this.getColumn(i).width = String(colWidths[i]) + " mm";
        }
    }

Be warned: This usually makes no sense except for trivial cases.

**Reply by 858882847** [reply]
What I mean is how to convert BIRT's column width output to Excel when exporting to Excel. What is the conversion ratio between them

**Reply by 858882847** [reply]
"If BIRT is exported as xlsx, the column width must be expressed in inches, otherwise the format context may be problematic, while Excel column width represents characters."
